Parser/OperatingSystem.php

```diff
@@ -238,43 +238,73 @@ public static function getShortOsData(string $name): array
     public function parse(): ?array
     {
         $return = $osRegex = $matches = [];
+        $name   = $version = $short = '';
 
-        foreach ($this->getRegexes() as $osRegex) {
-            $matches = $this->matchUserAgent($osRegex['regex']);
+        if ($this->clientHints && $this->clientHints->getOperatingSystem()) {
+            $hintName = $this->clientHints->getOperatingSystem();
 
-            if ($matches) {
-                break;
+            foreach (self::$operatingSystems as $osShort => $osName) {
+                if ($this->fuzzyCompare($hintName, $osName)) {
+                    $name  = $osName;
+                    $short = $osShort;
+
+                    break;
+                }
             }
-        }
 
-        if (empty($matches)) {
-            return $return;
-        }
+            $version = $this->clientHints->getOperatingSystemVersion();
 
-        $name                                = $this->buildByMatch($osRegex['name'], $matches);
-        ['name' => $name, 'short' => $short] = self::getShortOsData($name);
+            if ('Windows' === $name && '' !== $version) {
+                $majorVersion = (int) (\explode('.', $version)[0] ?? '0');
+                $version      = '7';
 
-        $version = \array_key_exists('version', $osRegex)
-            ? $this->buildVersion((string) $osRegex['version'], $matches)
-            : '';
+                if ($majorVersion > 0 && $majorVersion < 11) {
+                    $version = '10';
+                } elseif ($majorVersion > 10) {
+                    $version = '11';
+                }
+            }
+        }
 
-        foreach ($osRegex['versions'] ?? [] as $regex) {
-            $matches = $this->matchUserAgent($regex['regex']);
+        // parse the useragent if os wasn't provided in client hints
+        if (empty($name) || empty($version)) {
```

Comment:
not sure if we should check for the version here. There might be operating system that don't provide a version in the client hints. Not sure if we should parse the useragent as fallback in that case.
What do you think about parsing the useragent if no version is available in the client hints, but only use the version from useragent if the name matches?

Reply:
if major version got 0, it means that you don't need client prompts, see UserAgent (this only applies to windows)

Version | platformVersion
-- | --
Win7/8/8.1 | 0
Win10 1507 | 1
Win10 1511 | 2
Win10 1607 | 3
Win10 1703 | 4
Win10 1709 | 5
Win10 1803 | 6
Win10 1809 | 7
Win10 1903 | 8
Win10 1909 | 8
Win10 2004 | 10
Win10 20H2 | 10
Win10 21H1 | 10
Win10 21H2 | 10
Win11 | 13+

undocumented versions
Version | platformVersion
-- | --
win11 beta | 11 
win11 beta rc 1| 12 
win11 beta rc 2| 13